test/src/test/java/com/redhat/lightblue/test/metadata/FakeMetadataTest.java

```diff
@@ -36,6 +49,66 @@ public void testEntityInfo_VersionDoesExist(){
+            public DataStore parse(String name, MetadataParser<JsonNode> p, JsonNode node) {
+                return new DataStore() {
+                    @Override
+                    public String getBackend() {
+                        return "mongo";
+                    }
+                };
+            }
+
+            @Override
+            public void convert(MetadataParser<JsonNode> p, JsonNode emptyNode, DataStore object) {
+            }
+        });
+        JSONMetadataParser parser = new JSONMetadataParser(extensions, new DefaultTypes(), JsonNodeFactory.withExactBigDecimals(false));
+        JsonNode node = JsonUtils.json(getClass().getResourceAsStream("/usermd.json"));
+        EntityMetadata entityMetadata = parser.parseEntityMetadata(node);
+        metadata.setEntityMetadata(entityName, version1, entityMetadata);
+
+        JsonNode jsonSchema = metadata.getJSONSchema(entityName, version1);
+        String actual = jsonSchema.toString();
+        String expected = "{\"$schema\":\"http://json-schema.org/draft-04/schema#\",\"type\":\"object\",\"description\":\"JSON schema for entity 'user' version '1.0.0'\",\"properties\":{\"uid\":{\"type\":\"uid\"},\"nonrequid\":{\"type\":\"uid\",\"required\":\"Field not required constraint\"},\"iduid\":{\"type\":\"uid\",\"identity\":\"Field is part of identity constraint\"},\"personalInfo\":{\"type\":\"object\",\"lastName\":{\"type\":\"string\"},\"nonrequid\":{\"type\":\"uid\",\"required\":\"Field not required constraint\"},\"greeting\":{\"type\":\"string\"},\"department\":{\"type\":\"string\"},\"locale\":{\"type\":\"string\"},\"suffix\":{\"type\":\"string\"},\"title\":{\"type\":\"string\"},\"timezone\":{\"type\":\"string\"},\"faxNumber\":{\"type\":\"string\"},\"phoneNumber\":{\"type\":\"string\",\"matches\":\"^\\\\d{3}\\\\-\\\\d{4}\\\\ \\\\d{4}$\"},\"email\":{\"type\":\"string\"},\"company\":{\"type\":\"string\"},\"firstName\":{\"type\":\"string\"},\"requid\":{\"type\":\"uid\",\"required\":\"Field required constraint\"},\"emailConfirmed\":{\"type\":\"boolean\"}},\"updatedDate\":{\"type\":\"date\"},\"password\":{\"type\":\"object\",\"hashed\":{\"type\":\"string\"},\"salt\":{\"type\":\"string\"}},\"contactPermissions\":{\"type\":\"object\",\"allowMailContact\":{\"type\":\"boolean\"},\"allowThirdPartyContact\":{\"type\":\"boolean\"},\"allowPhoneContact\":{\"type\":\"boolean\"},\"allowFaxContact\":{\"type\":\"boolean\"},\"allowEmailContact\":{\"type\":\"boolean\"}},\"_id\":{\"type\":\"string\",\"identity\":\"Field is part of identity constraint\"},\"active\":{\"type\":\"boolean\"},\"login\":{\"type\":\"string\",\"maxLength\":\"64\",\"minLength\":\"1\",\"required\":\"Field required constraint\"},\"objectType\":{\"type\":\"string\"},\"requid\":{\"type\":\"uid\",\"description\":\"test\",\"required\":\"Field required constraint\"},\"createdDate\":{\"type\":\"date\"},\"required\":[\"login\",\"personalInfo.requid\",\"requid\"]}}";
+        JSONAssert.assertEquals(expected, actual, false);
```

Comment:
Should additionally verify that `actual` is a valid json schema.  We have a util that will [verify from a file resoruce](https://github.com/lightblue-platform/lightblue-core/blob/master/util/src/main/java/com/redhat/lightblue/util/JsonUtils.java#L122-L139).  Could update that class to have an api that supports passing in a json schema string for validation also.